media/mp4.ksy

```diff
@@ -0,0 +1,408 @@
+      - id: media_rate_integer
+        type: u2
+      - id: media_rate_fraction
+        type: u2
+        
+# -----------
+# -- MDHD --
+# -----------
+  box_body_mdhd:
+    seq:
+      - id: version
+        type: u4
+
+      - id: creation_time
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: modification_time
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: timescale
+        type: u4
+      - id: duration
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: lang_code #TODO: decode (or describe how to decode)
+        type: u2
+      - id: predefined
+        type: u2
+
+        
+# -----------
+# -- HDLR --
+# -----------
+  box_body_hdlr:
+    seq:
+      - id: predefined
+        type: u4
+      - id: reserved0
+        type: u4
+      - id: reserved1
+        type: u4
+      - id: reserved2
+        type: u4
+
+# -----------
+# -- VMHD --
+# -----------
+  box_body_vmhd:
+    seq:
+      - id: unknown
+        type: u4
+      - id: graphics_mode
+        type: u2
+      - id: reserved0
+        type: u2
+      - id: reserved1
+        type: u2
+      - id: reserved2
+        type: u2
+        
+# ----------
+# -- STSC --
+# ----------
+  box_body_stsc:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: box_body_stsc_entry
+        repeat: expr
+        repeat-expr: entry_count
+        
+  box_body_stsc_entry:
+    seq:
+      - id: first_chunk
+        type: u4
+      - id: samples_per_chunk
+        type: u4
+      - id: sample_description_index
+        type: u4
+
+        
+# ----------
+# -- STTS --
+# ----------
+  box_body_stts:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: box_body_stts_entry
+        repeat: expr
+        repeat-expr: entry_count
+        
+  box_body_stts_entry:
+    seq:
+      - id: sample_count
+        type: u4
+      - id: sample_delta
+        type: u4
+
+# ----------
+# -- STCO --
+# ----------
+  box_body_stco:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: chunk_offsets
+        type: u4
+        repeat: expr
+        repeat-expr: entry_count
+
+# ----------
+# -- CO64 --
+# ----------
+  box_body_co64:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: chunk_offsets
+        type: u8
+        repeat: expr
+        repeat-expr: entry_count
+
+# -----------
+# -- STSD --
+# -----------
+  box_body_stsd:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: box_body_stsd_entry
+        repeat: expr
+        repeat-expr: entry_count
+        
+  box_body_stsd_entry:
+    seq:
```

Comment:
For subtypes like these (i.e. which are clearly contained to one "box"), please consider using nested types, i.e.

```yaml
seq:
  # ...
  - id: entries
    type: entry
    repeat: expr
    repeat-expr: entry_count
types:
  entry:
# ...
```